shared/login/signup/success/index.js

```diff
@@ -25,9 +25,10 @@ Success.propTypes = {
 }
 
 export default connect(
-  state => ({paperkey: state.signup.paperkey}),
+  (state, ownProps) => ({
+    paperkey: ownProps.paperkey || state.signup.paperkey,
+    title: ownProps.title || state.signup.successTitle
+  }),
   dispatch => ({
-    onFinish: () => dispatch(sawPaperKey()),
```

Comment:
This'll probably break signup.  I wonder if there's a way to do something similar to the `(state, ownProps)` above but for dispatch.  We want to dispatch login's `sawPaperKey()` if we're in the login flow and signup's `sawPaperKey()` if we're in the signup flow.